src/controller/SetUpCodePairer.cpp

```diff
@@ -286,8 +289,21 @@ CHIP_ERROR SetUpCodePairer::StartDiscoveryOverWiFiPAF(SetupPayload & payload)
 CHIP_ERROR SetUpCodePairer::StopDiscoveryOverWiFiPAF()
 {
     mWaitingForDiscovery[kWiFiPAFTransport] = false;
+
 #if CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
-    DeviceLayer::ConnectivityMgr().WiFiPAFCancelIncompleteSubscribe();
+    if (mPAFSessionId != WiFiPAF::kUndefinedWiFiPafSessionId)
+    {
+        WiFiPAF::WiFiPAFSession sessionInfo = { .role = WiFiPAF::WiFiPafRole::kWiFiPafRole_Subscriber, .id = mPAFSessionId };
+        WiFiPAF::WiFiPAFLayer & pafLayer    = WiFiPAF::WiFiPAFLayer::GetWiFiPAFLayer();
+        WiFiPAF::WiFiPAFSession * pSession  = pafLayer.GetPAFInfo(WiFiPAF::PafInfoAccess::kAccSessionId, sessionInfo);
+        if ((pSession != nullptr) && (pSession->peer_id == WiFiPAF::kUndefinedWiFiPafSessionId))
+        {
+            // Canceling the subscription to discover the commissionee only if commissionee has not be discovered successfully.
+            // publish_id and subscribe_id are required in the follow-up packets
+            DeviceLayer::ConnectivityMgr().WiFiPAFCancelSubscribe(mPAFSessionId);
```

Comment:
OK, so say Session->peer_id  is not undefined, but this SetUpCodePairer is in fact going away.  What's going to prevent the PAF machinery from trying to use the `this` pointer it was passed, which is now pointing to deleted memory?

Again: we need an _unconditional_ "stop discovery" API, possibly one that says what exactly is stopping discovery, that will guarantee no callbacks after that API is called.  Otherwise we will have use-after-free security bugs here.